templates/todo/api/java/src/main/java/com/microsoft/azure/simpletodo/configuration/AzureDeveloperCliCredential.java

```diff
@@ -0,0 +1,220 @@
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.identity.CredentialUnavailableException;
+import com.azure.identity.implementation.IdentityClientOptions;
+import com.azure.identity.implementation.util.LoggingUtil;
+import com.azure.identity.implementation.util.ScopeUtil;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+import reactor.core.publisher.Mono;
+
+/**
+ * A credential provider that provides token credentials based on
+ * Azure Developer CLI command.
+ */
+@Immutable
+public class AzureDeveloperCliCredential implements TokenCredential {
+
+    private static final ClientLogger LOGGER = new ClientLogger(AzureDeveloperCliCredential.class);
+    private IdentityClientOptions identityClientOptions;
+
+    private static final String WINDOWS_STARTER = "cmd.exe";
+    private static final String LINUX_MAC_STARTER = "/bin/sh";
+    private static final String WINDOWS_SWITCHER = "/c";
+    private static final String LINUX_MAC_SWITCHER = "-c";
+    private static final String DEFAULT_WINDOWS_SYSTEM_ROOT = System.getenv("SystemRoot");
+    private static final String DEFAULT_MAC_LINUX_PATH = "/bin/";
+    private static final String WINDOWS_PROCESS_ERROR_MESSAGE = "'azd' is not recognized";
+    private static final Pattern LINUX_MAC_PROCESS_ERROR_MESSAGE = Pattern.compile("(.*)azd:(.*)not found");
+    private static final Pattern ACCESS_TOKEN_PATTERN = Pattern.compile("\"accessToken\": \"(.*?)(\"|$)");
+    private static final SerializerAdapter SERIALIZER_ADAPTER = JacksonAdapter.createDefaultSerializerAdapter();
+
+    /**
+     * Creates an AzureDeveloperCliCredential with default identity client options.
+     */
+    public AzureDeveloperCliCredential() {
+        identityClientOptions = new IdentityClientOptions();
+    }
+
+    @Override
+    public Mono<AccessToken> getToken(TokenRequestContext request) {
+        return authenticateWithAzureCli(request)
+            .doOnNext(token -> LoggingUtil.logTokenSuccess(LOGGER, request))
+            .doOnError(error -> LoggingUtil.logTokenError(LOGGER, identityClientOptions, request, error));
+    }
+
+    private Mono<AccessToken> authenticateWithAzureCli(TokenRequestContext request) {
+        StringBuilder azCommand = new StringBuilder("azd auth token --output json --scope ");
+
+        String scopes = ScopeUtil.scopesToResource(request.getScopes());
```

Comment:
I don't think we want to do this scope stuff - we should just pass the scopes as is (if there are multiple scopes, provide multiple `---scope` arguments).  This is about converting to an older format the the az CLI uses, `azd` is on the new plan.

Reply:
yes, missed that. Thank you.  Fixed it now